adobe_vipm/adobe/client.py

```diff
@@ -160,92 +161,119 @@ def create_customer_account(
-        response = requests.get(
-            urljoin(self._config.api_base_url, f"/v3/customers/{customer_id}/orders"),
-            headers=headers,
-            params={
-                "offer-id": product.sku,
-                "order-type": "NEW",
-                "status": STATUS_PROCESSED,
-            },
-        )
-
-        response.raise_for_status()
-        data = response.json()
-
-        if data["count"] > 0:
-            return data["items"][0]
-
-    @wrap_http_error
-    def search_last_return_order_by_order(
-        self,
-        reseller_country: str,
-        customer_id: str,
-        adobe_order_id: str,
-    ) -> dict | None:
-        """
-        Search for an order of type RETURN by the identifier of the returned order.
-        The `reseller_country` is used to select the reseller and the Adobe credentials
-        of the account to which the reseller belong to.
 
-        Args:
-            reseller_country (str): The country of the reseller to which the customer account
-            belongs to.
-            customer_id (str): Identifier of the customer that placed the order.
-            adobe_order_id (str): Identifier of the order that should have been returned.
+        orders = []
+        orders_base_url = f"/v3/customers/{customer_id}/orders"
 
-        Returns:
-            dict: The RETURN order or `None` if not found.
-        """
-        reseller: Reseller = self._config.get_reseller(reseller_country)
-        headers = self._get_headers(reseller.distributor.credentials)
-        response = requests.get(
-            urljoin(self._config.api_base_url, f"/v3/customers/{customer_id}/orders"),
-            headers=headers,
-            params={
-                "reference-order-id": adobe_order_id,
-                "order-type": "RETURN",
-                "status": (STATUS_PROCESSED, STATUS_PENDING),
-            },
-        )
+        new_orders_params = {
+            "order-type": ORDER_TYPE_NEW,
+            "status": STATUS_PROCESSED,
+            "limit": 100,
+            "offset": 0,
+        }
 
-        response.raise_for_status()
-        data = response.json()
+        new_orders_next_url = f"{orders_base_url}?{urlencode(new_orders_params)}"
 
-        if data["count"] > 0:
-            return data["items"][0]
+        while new_orders_next_url:
+            new_orders_response = requests.get(
+                urljoin(self._config.api_base_url, new_orders_next_url),
+                headers=headers,
+            )
+            new_orders_response.raise_for_status()
+            new_orders_page = new_orders_response.json()
+            for order in new_orders_page["items"]:
+                actual_sku = get_actual_sku(order["lineItems"], sku)
+                if not actual_sku:
+                    continue
+
+                logger.debug(f"Found order to return for sku {actual_sku}: {order['orderId']}")
+
+                item_to_return = next(
```

Comment:
I guess there is an util function `find_first`?

Reply:
yeah see below.